library/core/class.uploadimage.php

```diff
@@ -323,12 +323,11 @@ public static function saveImageAs($Source, $Target, $Height = '', $Width = '',
      * @param $GD
      * @param $TargetPath
      */
-    public static function imageIco($GD, $TargetPath) {
-        require_once PATH_LIBRARY.'/vendors/phpThumb/phpthumb.ico.php';
-        require_once PATH_LIBRARY.'/vendors/phpThumb/phpthumb.functions.php';
-        $Ico = new phpthumb_ico();
-        $Arr = array('ico' => $GD);
-        $IcoString = $Ico->GD2ICOstring($Arr);
-        file_put_contents($TargetPath, $IcoString);
+    public static function imageIco($gd, $targetPath) {
+        $imagePath = tempnam(sys_get_temp_dir(), 'iconify');
+        imagepng($gd, $imagePath);
+
+        $icoLib = new PHP_ICO($imagePath, [[16,16]]);
+        $icoLib->save_ico($targetPath);
```

Comment:
We're creating a temporary image file to hand off to `PHP_ICO`, then writing the icon to `$targetPath`.  Can we delete the file at `$imagePath` after we're done with it, instead of hoping the system will properly maintain the temp directory?